index.bs

```diff
@@ -313,19 +349,13 @@ The abstract operation GetRegistryEntry with arguments registry and key performs
 1. Let _pair_ be the entry in _entries_ such that _pair_.[[key]] is equal to _key_.
 1. If _pair_ does not exist, then return *null*.
 1. Let _entry_ be _pair_.[[value]].
+1. Let _currentStage_ be GetCurrentStage(_entry_).
+1. Let _result_ be the result of transforming _currentStage_.[[Result]] with a new pass-through promise.
+1. Let _stage_ be _currentStage_.[[Stage]].
 1. Let _result_ be CreateObject().
-1. Call SimpleDefine(_result_, "state", _entry_.[[State]]).
-1. Let _statePromise_ be *undefined*.
-1. If _entry_.[[State]] is "fetch" and _entry_.[[Fetch]] is not *undefined*, then
-  1. Set _statePromise_ to the result of transforming _entry_.[[Fetch]] with a new pass-through promise.
-1. Else If _entry_.[[State]] is "translate" and _entry_.[[Translate]] is not *undefined*, then
-  1. Set _statePromise_ to the result of transforming _entry_.[[Translate]] with a new pass-through promise.
-1. Else If _entry_.[[State]] is "instantiate" and _entry_.[[Instantiate]] is not *undefined*, then
-  1. Set _statePromise_ to the result of transforming _entry_.[[Instantiate]] with a fulfillment handler that, when called with argument _entry_, runs the following steps:
-    1. If _entry_.[[Module]] is a Function object, then return _entry_.[[Module]].
-    1. Return *undefined*.
-1. Call SimpleDefine(_result_, "statePromise", _statePromise_).
-1. If _entry_.[[State]] is "ready" then let _module_ be _entry_.[[Module]].
+1. Call SimpleDefine(_result_, "stage", _stage_).
```

Comment:
Question: right now we are exposing `stage` and `stagePromise` as part of the shim for the registry entry. I think we can make this `stage` and `value`, which is just the thenable value for the entry's "ready" stage, or the current stage.